modules/core/04-channel/types/upgrade.go

```diff
@@ -117,3 +117,12 @@ func (u *UpgradeError) GetErrorReceipt() ErrorReceipt {
 		Message:  fmt.Sprintf("ABCI code: %d: %s", code, restoreErrorString),
 	}
 }
+
+// IsUpgradeError returns true if err is of type UpgradeErroe, otherwise false.
+func IsUpgradeError(err error) bool {
+	if err == nil {
+		return false
+	}
+	_, ok := err.(*UpgradeError)
+	return ok
```

Comment:
This should also do:

```suggestion
	_, ok := err.(*UpgradeError)
	return ok
```
type assertions check that the arg isn't nil.